src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessFetch.cs

```diff
@@ -52,6 +52,16 @@ public void Process(FetchRequestBase resultOperator, QueryModelVisitor queryMode
 				{
 					var metadata = queryModelVisitor.VisitorParameters.SessionFactory
 													.GetClassMetadata(resultOperator.RelationMember.ReflectedType);
+					if (metadata == null)
+					{
+						var entityName = queryModelVisitor.VisitorParameters.SessionFactory.GetImplementors(
+							resultOperator.RelationMember.ReflectedType.FullName).SingleOrDefault();
```

Comment:
Is there a risk to have many implementors, so a crash here, in case of inheritance mapping?

Reply:
For some reasons I didn't think it's a valid scenario. Fixed.